modules/library/jdbc/src/main/java/org/geotools/jdbc/SQLDialect.java

```diff
@@ -1463,4 +1468,18 @@ protected PrimaryKey getPrimaryKey(String typeName) throws IOException {
         SimpleFeatureType featureType = dataStore.getSchema(typeName);
         return dataStore.getPrimaryKey(featureType);
     }
+    
+    protected void encodeFunction( String function, String column, StringBuffer sql ) {
```

Comment:
Functions in general have several parameters. See also https://github.com/geotools/geotools/blob/870e3ab8eece49c54186241e74adf6645b925f1c/modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/FilterToSqlHelper.java#L536 for existing support on encoding filter functions (not aggregate functions).